src/models/converters/transfers.rs

```diff
@@ -1,59 +1,175 @@
-use crate::models::service::transactions::{TransferInfo, TransactionInfo, Erc20Transfer, Erc721Transfer, EtherTransfer};
+use crate::models::backend::transfers::{
+    Erc20TokenInfo, Erc20Transfer as Erc20TransferDto, Erc721TokenInfo,
+    Erc721Transfer as Erc721TransferDto, EtherTransfer as EtherTransferDto,
+    Transfer as TransferDto,
+};
+use crate::models::service::transactions::details::TransactionDetails;
 use crate::models::service::transactions::Transfer as ServiceTransfer;
-use crate::models::backend::transfers::{Transfer as TransferDto, Erc20Transfer as Erc20TransferDto, Erc721Transfer as Erc721TransferDto, EtherTransfer as EtherTransferDto};
+use crate::models::service::transactions::{
+    Erc20Transfer, Erc721Transfer, EtherTransfer, TransactionInfo, TransactionStatus, TransferInfo,
+};
+use crate::providers::info::{
+    InfoProvider, TokenInfo, TokenType,
+};
+use anyhow::Result;
 
 impl TransferDto {
-    pub fn to_transfer(&self) -> TransactionInfo {
+    pub fn to_transfer(&self, info_provider: &mut InfoProvider) -> TransactionInfo {
         match self {
-            TransferDto::Erc721(transfer) => TransactionInfo::Transfer(transfer.to_transfer_transaction()),
-            TransferDto::Erc20(transfer) => TransactionInfo::Transfer(transfer.to_transfer_transaction()),
-            TransferDto::Ether(transfer) => TransactionInfo::Transfer(transfer.to_transfer_transaction()),
-            _ => TransactionInfo::Unknown
+            TransferDto::Erc721(transfer) => {
+                TransactionInfo::Transfer(transfer.to_transfer_transaction(info_provider))
+            }
+            TransferDto::Erc20(transfer) => {
+                TransactionInfo::Transfer(transfer.to_transfer_transaction(info_provider))
+            }
+            TransferDto::Ether(transfer) => {
+                TransactionInfo::Transfer(transfer.to_transfer_transaction())
+            }
+            _ => TransactionInfo::Unknown,
+        }
+    }
+
+    pub fn to_transaction_details(
+        &self,
+        info_provider: &mut InfoProvider,
+    ) -> Result<TransactionDetails> {
+        Ok(TransactionDetails {
+            executed_at: self.get_execution_time(),
+            submitted_at: None,
+            tx_status: TransactionStatus::Success,
+            tx_info: self.to_transfer(info_provider),
+            tx_data: None,
+            tx_hash: self.get_transaction_hash(),
```

Comment:
This feels like it should be part of the `executionInfo`. Maybe I am misunderstanding here what's what.

Reply:
every type of transaction can have a transaction hash (of the ethereum transaction that contained it). Only case were it is not always present is for multisig transactions that have not been executed. Therefore I would have it here as it would be a field that is present for all types.